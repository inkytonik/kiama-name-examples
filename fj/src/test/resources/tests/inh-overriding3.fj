class A extends Object {
  A() {
    super();
  }
  A mk(X x) {
    return new A();
  }
}
class B extends A {
  B() {
    super();
  }
  A mk(Y y) {
    return new B();
  }
}
class X extends Object {
  X() {
    super();
  }
}
class Y extends X {
  Y() {
    super();
  }
}
new B().mk(new Y())
