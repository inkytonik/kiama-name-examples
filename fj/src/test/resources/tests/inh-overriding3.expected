src/test/resources/tests/inh-overriding3.fj:5:10:info: x def: X
  A mk(X x) {
         ^
src/test/resources/tests/inh-overriding3.fj:6:12:info: new A(): A
    return new A();
           ^
src/test/resources/tests/inh-overriding3.fj:13:3:error: method type (Y) : A does not match overridden type (X) : A
  A mk(Y y) {
  ^
src/test/resources/tests/inh-overriding3.fj:13:10:info: y def: Y
  A mk(Y y) {
         ^
src/test/resources/tests/inh-overriding3.fj:14:12:info: new B(): B
    return new B();
           ^
src/test/resources/tests/inh-overriding3.fj:27:1:info: new B().mk ...: A
new B().mk(new Y())
^
src/test/resources/tests/inh-overriding3.fj:27:1:info: new B(): B
new B().mk(new Y())
^
src/test/resources/tests/inh-overriding3.fj:27:12:info: new Y(): Y
new B().mk(new Y())
           ^
A
